src/lib/dnssd/minimal_mdns/core/QName.cpp

```diff
@@ -152,6 +152,40 @@ bool SerializedQNameIterator::operator==(const FullQName & other) const
     return ((idx == other.nameCount) && !self.Next());
 }
 
+bool SerializedQNameIterator::operator==(const SerializedQNameIterator & other) const
+{
+    SerializedQNameIterator a = *this; // allow iteration
+    SerializedQNameIterator b = other;
+
+    while (true)
+    {
+        bool hasA = a.Next();
+        bool hasB = b.Next();
+
+        if (hasA ^ hasB)
+        {
+            return false; /// one is longer than the other
```

Comment:
Not sure how a logical exclusive-or on `hasA ^ hasB` can mean `one is longer than the other`.

Reply:
One has next, the other does not. So if not both are false or both are true, this means one is longer than  the other.
Not sure how to else explain it. At least unit tests should validate this a lot.